app/controllers/application_controller.rb

```diff
@@ -18,6 +18,9 @@ def set_locale
       # { :locale => I18n.locale }
   # end
 
+  rescue_from(ActionController::ParameterMissing) do |e|
+    render text: "Request is missing param '#{e.param}'", status: :bad_request
```

Comment:
What about to use one `rescue_from` for API and return text in requested format (JSON, YAML) and one here rendering just test?

Also is there any reason why text is not part of locales?